azure-stack/hci/manage/manage-syslog-forwarding.md

```diff
@@ -0,0 +1,469 @@
+ 
+   $domainAdminCred = New-Object System.Management.Automation.PSCredential ($domainAdmin, $domainAdminPassword) 
+ 
+   $certPassword = ConvertTo-SecureString -String "<your client certificate password>" -AsPlainText -Force 
+   $clientCertPath = "<local file path to your client cert pfx file on DVM>" 
+ 
+   Import-Module C:\CloudDeployment\ECEngine\CloudEngine.dll 
+
+   Import-Module C:\CloudDeployment\ECEngine\TestEceInterface.psm1 
+
+   Import-Module C:\CloudDeployment\ECEngine\EnterpriseCloudEngine.psd1 
+
+   $Parameters = Get-EceInterfaceParameters -RolePath "BareMetal" -InterfaceName Deployment 
+
+   $nodes = Get-ClusterNode -Cluster “s-cluster” | ForEach-Object Name
+   ```
+
+1. Import client certificate for ($node in $nodes).
+
+   ```azurepowershell
+   { 
+   $params = @{  
+      ComputerName = $nodeName  
+      Credential = $domainAdminCred 
+   } 
+ 
+   $session = New-PSSession @params 
+ 
+   $tempPath = Invoke-Command -Session $session -ScriptBlock { 
+    return $env:TEMP 
+   } 
+   $clientCertPathOnNode = "$env:Temp\client.pfx" 
+   Copy-Item -ToSession $session -Path $clientCertPath -Destination $clientCertPathOnNode 
+ 
+   $params = @{  
+      Session = $session  
+      ArgumentList = @($serverName, $serverPort, $clientCertPathOnNode, $certPassword)  
+   } 
+   Invoke-Command @params -ScriptBlock {  
+    param($ServerName, $ServerPort, $certPath, $certPassword) 
+ 
+   Import-PfxCertificate -FilePath $certPath -CertStoreLocation "Cert:\\LocalMachine\My" -Password $certPassword 
+    Remove-Item -Path $certPath 
+   } 
+  
+   Remove-PSSession -Session $session 
+   } 
+ 
+   Import self-signed root certificate
+   ```
+
+   Use the following section only in relevant cases:
+
+   ```azurepowershell
+   $rootCertPath = "local file path to your client cert pfx file on DVM" 
+   foreach ($node in $nodes) 
+   { 
+   $params = @{  
+      ComputerName = $nodeName  
+      Credential = $domainAdminCred 
+   }
+
+   $session = New-PSSession @params
+
+   $tempPath = Invoke-Command -Session $session -ScriptBlock { 
+    return $env:TEMP 
+   } 
+  
+   $rootCertPathOnNode = "$env:Temp\root.pfx" 
+   Copy-Item -ToSession $session -Path $rootCertPath -Destination $rootCertPathOnNode 
+
+   $params = @{  
+      Session = $session  
+      ArgumentList = @($serverName, $serverPort, $clientCertPathOnNode)  
+   } 
+  
+   Invoke-Command @params -ScriptBlock {  
+   param($ServerName, $ServerPort, $certPath) 
+
+   Import-PfxCertificate -FilePath $certPath -CertStoreLocation "Cert:\\LocalMachine\root" -Password $certPassword 
+   Remove-Item -Path $certPath 
+   } 
+   Remove-PSSession -Session $session 
+   }
+   ```
+
+1. Set the syslog forwarder to use mutual authentication.
+
+   An action plan will be started and its ID will be stored in `$actionPlanInstanceId`. Monitor the action plan and validate that it completes successfully.
+
```

Comment:
Tweak as per the style guidelines: This script starts the action plan and the action plan ID is stored in ......

Reply:
Done > revised.